### PR TITLE
br: Fix new collaction enable check (#33500)

### PR DESCRIPTION
**Pull request by ti-srebot**
cherry-pick #33500 to release-5.2
You can switch your code base to this Pull Request by using [git-extras](https://github.com/tj/git-extras):
```bash
# In tidb repo:
git pr https://github.com/pingcap/tidb/pull/33534
```

After apply modifications, you can push your change to this PR via:
```bash
git push git@github.com:ti-srebot/tidb.git pr/33534:release-5.2-8f8dfafc105d
```

---

### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/33422

Problem Summary:

### What is changed and how it works?
- backup up-stream `NewCollactionEnable` into backupmeta
- check `NewCollactionEnable` in backupmeta is the same as the down-stream

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
Fix the issue that NewCollationEnable config not checked during restoration.
```

### CODE REVIEW

**Comment by joccau** [comment]
### Manual test (add detailed scripts or steps below)

1. test it using command

- run command
`
./bin/br backup full -u $IP:$port -s "s3://tmp/full?access-key=minioadmin&secret-access-key=minioadmin&endpoint=http://$IP:9000&force-path-style=true"
`
- to find log
`
grep "newCollationEnable" /var/folders/37/dyx4gvk55hx8wvl6j4mpr81w0000gn/T/br.log.2022-04-13T18.00.20+0800
`
- the log is existed
`
[2022/04/13 18:00:21.148 +08:00] [INFO] [backup.go:261] ["get newCollationEnable for check during restore"] [newCollationEnable=True]
`

File: br/pkg/task/backup.go [hunk 1]
```diff
@@ -250,6 +250,16 @@ func RunBackup(c context.Context, g glue.Glue, cmdName string, cfg *BackupConfig
...
 
+	se, err := g.CreateSession(mgr.GetStorage())
+	if err != nil {
+		return errors.Trace(err)
+	}
+	newCollationEnable, err := se.GetGlobalVariable(tidbNewCollationEnabled)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	log.Info("get newCollationEnable for check during restore", zap.String("newCollationEnable", newCollationEnable))
```
Comment (3pointer) [inline br/pkg/task/backup.go:261]: please refine the name.

Comment (joccau) [reply br/pkg/task/backup.go]: > 

Done

**Review by 3pointer — APPROVED** [review]
LGTM

**Review by YuJuncen — APPROVED** [review]

**Comment by VelocityLight** [comment]
This pull request is closed because it's related version has closed automatic cherry-picking.
If it's still needed, you can reopen it or just regenerate it using bot,
see:
- https://prow.tidb.io/command-help#cherrypick
- https://book.prow.tidb.net/#/plugins/cherrypicker

You can find more details at:
- https://internals.tidb.io/t/topic/785

**Pull request closed without merging**